File: 1666-make-the-string-great/make-the-string-great.go
func makeGood(s string) string {
    var diff byte
    diff = 'a' - 'A'
    prevlength := -1
    for ; prevlength != len(s) ;{
        prevlength = len(s)
        for i := 0 ; i < len(s) - 1 ; i++ {
            currentDiff := s[i+1] - s[i]
            if currentDiff == diff || currentDiff + diff == 0 {
                s = s[0:i] + s[i+2:len(s)]
            }
        }
    }
    return s
}

File: 1666-make-the-string-great/make-the-string-great.py
class Solution:
    def makeGood(self, s: str) -> str:
        diff = abs(ord('A') - ord('a'))
        i = 0
        while i < len(s)-1:
            currDiff = abs(ord(s[i+1]) - ord(s[i]))
            if diff == currDiff:
                s = s[:i] + s[i+2:]
                if i!=0 :
                    i-=1
                continue
            i+=1
        
        return s

        